src/main/kotlin/no/item/xp/plugin/writer/TypeScriptWriter.kt

```diff
@@ -1,3 +1,68 @@
 package no.item.xp.plugin.writer
 
-class TypeScriptWriter
+import java.io.File
+import no.item.xp.plugin.models.GeneratedField
+import no.item.xp.plugin.models.InputType
+import no.item.xp.plugin.models.XmlType
+import java.io.BufferedWriter
+
+@Suppress("FunctionName")
+fun TypeScriptWriter(interfaceName: String, type: XmlType, objList: List<GeneratedField>?): String {
+  var content = "export interface "
+  content += xmlTypeFormatted(type) + "{\n"
+  if (objList != null) {
+    for (fields: GeneratedField in objList) {
+      content += returnCommentFromLabel(fields)
+      content += returnStringValue(fields)+"\n"
+      content += "\n"
+    }
+  }
+  content+="}"
+  File(interfaceName).bufferedWriter().use { out: BufferedWriter ->
+    out.write(content)
+  }
+  return interfaceName
+}
+
+fun returnStringValue(fields: GeneratedField): String {
+  var returnString = ""
+  returnString+=fields.name
+  returnString+= isNullable(fields.nullable)
+  returnString+=": "
+  returnString+=typeScriptType(fields.type)
+  returnString+=";"
+  return returnString
+}
+
+fun typeScriptType(type: InputType): String {
+  return when(type){
+    InputType.CHECKBOX -> "boolean"
+    InputType.DOUBLE -> "number"
+    InputType.LONG -> "number"
+    else -> "string"
+  }
+}
+
+fun isNullable(bol: Boolean):String{
+  return if (!bol) "" else "?"
+}
+fun xmlTypeFormatted(type: XmlType): String {
+  return when(type){
+    XmlType.SITE -> "Site"
+    XmlType.ID_PROVIDER -> "IdProvider"
+    XmlType.CONTENT_TYPE -> "ContentType"
+    XmlType.LAYOUT -> "Layout"
+    XmlType.PART -> "Part"
+    XmlType.PAGE -> "Page"
+    XmlType.MACRO -> "Macro"
+    XmlType.MIXIN -> "Mixin"
+  }
+}
+
+fun returnCommentFromLabel(fields: GeneratedField): String {
+  return if(fields.comment!=null){
+    "/** " + fields.comment + " */\n"
+  } else{
+    "\n"
+  }
```

Comment:
Try to express this using `Option` to handle the `null` case

Reply:
We changed comment in GeneratedField not to be nullable. So that kinda goes out, but did a change to check for empty field.